apps/condo/domains/acquiring/schema/CreatePaymentByLinkService.test.js

```diff
@@ -371,6 +371,55 @@ describe('CreatePaymentByLinkService', () => {
         expect(payments[0].receipt).toBeNull()
     })
 
+    test('user: create virtual multiPayment if no receipts found and no PaymPeriod passed', async () => {
+        const {
+            organization,
+            qrCode: qr,
+            qrCodeAttrs,
+        } = await createOrganizationAndPropertyAndQrCode(admin, 14, 4, '07.2023')
+
+        delete qrCodeAttrs['PaymPeriod']
+        const qrCode = qr.replace(/\|PaymPeriod=[^|]*/, '')
+
+        const { billingIntegrationContext } = await addBillingIntegrationAndContext(admin, organization, {}, { status: CONTEXT_FINISHED_STATUS, settings: { dv: 1, receiptUploadDate: 20 } })
+        const recipient = createTestRecipient({
+            name: organization.name,
+            tin: organization.tin,
+            bic: qrCodeAttrs.BIC,
+            bankAccount: qrCodeAttrs.PersonalAcc,
+        })
+        const { acquiringIntegration } = await addAcquiringIntegrationAndContext(admin, organization, {}, { status: CONTEXT_FINISHED_STATUS, recipient })
+
+        const [billingProperty] = await createTestBillingProperty(admin, billingIntegrationContext)
+        await createTestBillingAccount(admin, billingIntegrationContext, billingProperty, { number: qrCodeAttrs.PersAcc })
+        await createTestBillingRecipient(admin, billingIntegrationContext, { bankAccount: qrCodeAttrs.PersonalAcc })
+        await createTestBillingRecipient(admin, billingIntegrationContext)
+        const [bankAccount] = await createTestBankAccount(admin, organization, {
+            number: qrCodeAttrs.PersonalAcc,
+            routingNumber: qrCodeAttrs.BIC,
+        })
+
+        const [data] = await createPaymentByLinkByTestClient(user, { qrCode })
+
+        expect(data.address).toBeDefined()
+        expect(data.accountNumber).toEqual(qrCodeAttrs.PersAcc)
+        expect(data.multiPaymentId).toBeDefined()
+        expect(data.unitName).toBeDefined()
+        expect(data.acquiringIntegrationHostUrl).toBe(acquiringIntegration.hostUrl)
+        expect(data.currencyCode).toBe(billingIntegrationContext.integration.currencyCode)
+
+        const multiPayment = await MultiPayment.getOne(admin, { id: data.multiPaymentId })
+        expect(multiPayment).toBeDefined()
+
+        const payments = await Payment.getAll(admin, {
+            multiPayment: { id: multiPayment.id },
+        })
+
+        expect(payments[0].accountNumber).toBe(qrCodeAttrs.PersAcc)
```

Comment:
`expect(payments.length).toBe(0)` or use `.getOne`?

Reply:
The checking for array length was added